tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,521 @@
+    if($relPath){
+        try {
+            Push-Location $relPath
+            $relPath = Resolve-Path -Relative -Path $discoveredPath
+
+            # relpaths are returned with ".\<blah>"
+            # given that, we need to get rid of it. This has possiiblity for bugs down the line.
+            $relPath = $relPath.Substring(2)
+        }
+        finally {
+            Pop-Location
+        }
+
+        # relative path to assets Json from within path
+        Add-Member -InputObject $config -MemberType "NoteProperty" -Name "AssetsJsonRelativeLocation" -Value $relPath
+    }
+
+    return $config
+}
+
+Function Resolve-AssetStore-Location {
+    if (-not (Test-Path $ASSETS_STORE)){
+        mkdir -p $ASSETS_STORE | Out-Null
+    }
+    $ASSETS_STORE = Resolve-Path $ASSETS_STORE
+
+    return $ASSETS_STORE
+}
+
+Function Get-MD5-Hash {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Input
+    )
+
+    $stringAsStream = [System.IO.MemoryStream]::new()
+    $writer = [System.IO.StreamWriter]::new($stringAsStream)
+    $writer.write($Input)
+    $writer.Flush()
+    $stringAsStream.Position = 0
+    return Get-FileHash -InputStream $stringAsStream -Algorithm MD5
+}
+
+Function Resolve-AssetRepo-Location {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config
+    )
+    $assetsLocation = Resolve-AssetStore-Location
+    $repoName = $Config.AssetsRepo.Replace("/", ".")
+
+    # this is where we will need to handle the multi-copying of the repository.
+    # to begin with, we will use the relative path of the assets json in combination with the
+    # Repo/RepoId to create a unique hash. In the future, we will need to take the targeted commit into account,
+    # and resolve conflicts
+    if ($Config.AssetsRepoId) {
+        $repoName = $Config.AssetsRepoId
+    }
+
+    $repoNameHashed = Get-MD5-Hash -Input ((Join-Path $repoName $Config.AssetsJsonRelativeLocation).ToString())
+
+    $repoPath = (Join-Path $assetsLocation $repoNameHashed.Hash)
+    
+    if (-not (Test-Path $repoPath)){
+        mkdir -p $repoPath | Out-Null
+    }
+    
+    return $repoPath
+}
+
+Function Get-Default-Branch {
+    param(
+        $Config
+    )
+    $repoJsonResult = curl "https://api.github.com/repos/$($Config.AssetsRepo)"
+    return ($repoJsonResult | ConvertFrom-Json).default_branch
+}
+
+<#
+.SYNOPSIS
+This function returns a boolean that indicates whether or not the assets repo has been initialized.
+#>
+Function Is-AssetsRepo-Initialized {
+    param(
+        [PSCustomObject] $Config
+    )
+
+    $result = $false
+    $assetRepoLocation = Resolve-AssetRepo-Location -Config $Config
+
+    try {
+        Push-Location $assetRepoLocation
+        $gitLocation = Join-Path $assetRepoLocation ".git"
+
+        if (Test-Path $gitLocation){
+            $result = $true
+        }
+        else{
+            $result = $false
+        }
+    }
+    catch {
+        Write-Error $_
```

Comment:
What are the expected possible errors here? That the `Push-Location` would fail? I think this is another case where handling specific cases might be easier and more clear, but maybe I just like coding in go too much 🤷